src/main/java/org/jnode/fs/xfs/XfsDirectory.java

```diff
@@ -91,7 +91,7 @@ protected FSEntryTable readEntries() throws IOException {
                         break;
                     }
 
-                    INode childInode = fileSystem.getInodeBTree().getINode(entry.getINumber());
+                    INode childInode = fileSystem.getINode(entry.getINumber());
```

Comment:
Will `INodeBTree` eventually be used?
The implementation of `XfsFileSystem.getINode()` vs `INodeBTree.getINode()` differ quite a bit. I'll need to read a bit more into it.

Reply:
The INodeBtree will not be used, the code was adapted to use XfsFileSystem to implement the getINode function according with the XFS specifications.